.github/workflows/inspektor-gadget.yml

```diff
@@ -1282,6 +1282,34 @@ jobs:
           BUILDER_IMAGE=${{ needs.build-helper-images.outputs.ebpf_builder_image }} \
           push-gadgets -o install/ig
 
+  image-based-gadgets-test:
+    name: Image-based gadgets test
+    #level 2
+    needs:
+      - build-ig
+      - build-helper-images
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Get ig-linux-amd64.tar.gz from artifact.
+      uses: actions/download-artifact@v3
+      with:
+        name: ig-linux-amd64-tar-gz
+        path: /home/runner/work/inspektor-gadget/
+    - name: Unpack ig-linux-amd64.tar.gz
+      run: |
+        tar zxvf /home/runner/work/inspektor-gadget/ig-linux-amd64.tar.gz
+        sudo mv ig /usr/bin/ig
```

Comment:
Build-ig builds ig and adds it as an artifact.
```
- name: Build ig-linux-${{ matrix.platform }}
      run: |
        if [ "${{ matrix.platform }}" = 'arm64' ]; then
          sudo apt-get update
          sudo apt-get install qemu-user-static
        fi

        make EBPF_BUILDER=${{ needs.build-helper-images.outputs.ebpf_builder_image }} ig-linux-${{ matrix.platform }}

        # Prepare assets for release and actions artifacts
        mkdir ${{ matrix.platform }}
        cp ig-linux-${{ matrix.platform }} ${{ matrix.platform }}/ig
        cp LICENSE ${{ matrix.platform }}/
        tar --sort=name --owner=root:0 --group=root:0 \
          -czf ig-linux-${{ matrix.platform }}.tar.gz -C ${{ matrix.platform }} \
          ig LICENSE
        rm -rf ${{ matrix.platform }}
    - name: Add ig-linux-${{ matrix.platform }}.tar.gz as artifact.
      uses: actions/upload-artifact@v3
      with:
        name: ig-linux-${{ matrix.platform }}-tar-gz
        path: /home/runner/work/inspektor-gadget/inspektor-gadget/ig-linux-${{ matrix.platform }}.tar.gz
```

Reply:
> Are those steps necessary? Wouldn't the `ig` file already placed on the filesystem after the `build-ig` job? I see some other jobs follow the same pattern, so I am probably wrong, but I'd like to understand.

It's because each job runs in a different runner environment, see [documentation](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#overview). So, they don't share the filesystem.

Reply:
> Are those steps necessary? Wouldn't the `ig` file already placed on the filesystem after the `build-ig` job? I see some other jobs follow the same pattern, so I am probably wrong, but I'd like to understand.

It's because each job runs in a different runner environment, see [documentation](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#overview). So, they don't share the filesystem.